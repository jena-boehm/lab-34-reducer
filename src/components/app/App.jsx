/* eslint-disable max-len */
import React, { useReducer } from 'react';
import colorReducer, { initialState } from '../reducers/colorReducer';

function App() {
  const [state, dispatch] = useReducer(colorReducer, initialState);
  const { before, current, after } = state;

  const undo = () => {
    dispatch({ type: 'UNDO' });
  };

  const redo = () => {
    dispatch({ type: 'REDO' });
  };

  return (
    <>
      <button 
        data-testid="undoButton"
        id="UNDO"
        onClick={undo}
        disabled={!before.length}>undo</button>
      <button 
        data-testid="redoButton"
        id="REDO"
        onClick={redo}
        disabled={!after.length}>redo</button>

      <input 
        data-testid="colorInput"
        id="RECORD"
        type="color" 
        value={current} 
        onChange={({ target }) => dispatch({
          payload: target.value,
          type: target.id
        })} 
      />

      <div 
        data-testid="colorDiv"
        style={{ 
          backgroundColor: current, 
          width: '10rem', 
          height: '10rem' }}>
      </div>
    </>
  );
}

export default App;
