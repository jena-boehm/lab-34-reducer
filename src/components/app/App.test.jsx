import React from 'react';
import { render, cleanup, screen, fireEvent } from '@testing-library/react';
import App from './App';

describe('App component', () => {
  afterEach(() => cleanup());
  
  it('renders App', () => {
    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('changes the background color', async() => {
    render(<App />);
    const colorInput = screen.getByLabelText('Color Input');
    fireEvent.change(colorInput, {
      target: {
        value: '#00FF00'
      }
    });

    const colorDiv = await screen.getByTestId('colorDiv');
    expect(colorDiv).toHaveStyle({ backgroundColor: '#00FF00' });
  });
});

