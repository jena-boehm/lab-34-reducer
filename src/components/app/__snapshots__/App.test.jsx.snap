// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders App 1`] = `
<DocumentFragment>
  <button
    data-testid="undoButton"
    disabled=""
    id="UNDO"
  >
    undo
  </button>
  <button
    data-testid="redoButton"
    disabled=""
    id="REDO"
  >
    redo
  </button>
  <input
    data-testid="colorInput"
    id="RECORD"
    type="color"
    value="#FF0000"
  />
  <div
    data-testid="colorDiv"
    style="background-color: rgb(255, 0, 0); width: 10rem; height: 10rem;"
  />
</DocumentFragment>
`;
